cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -470,6 +484,17 @@ private boolean isConcurrentRunsAllowed(ProgramType type) {
     return null;
   }
 
+  private List<ProgramRuntimeService.RuntimeInfo> findRuntimeInfos(ProgramId programId) {
+    Map<RunId, ProgramRuntimeService.RuntimeInfo> runtimeInfos = runtimeService.list(programId.getType());
```

Comment:
It can be as simple as `runtimeService.list(programId.toId()).values()`, meaning you don't even need to introduce this method.